flan/v2/task_configs_xp3.py

```diff
@@ -0,0 +1,132 @@
+    elif isinstance(feature, list):
+        return [feature_to_spec(f, length=length) for f in feature]
+    elif isinstance(feature, dict):
+        return {k: feature_to_spec(v, length=length) for k, v in feature.items()}
+    else:
+        raise ValueError(f"Unparseable feature type {type(feature)}")
+
+def hf_dataset_to_tf_dataset(dataset):
+    return tf.data.Dataset.from_generator(
+        dataset.__iter__, output_signature={k: feature_to_spec(v) for k, v in dataset.features.items()}
+    )
+
+def get_tf_dataset(split, shuffle_files, dataset_name, subset_name, split_mapping,seed):
+    # HF datasets does not support file-level shuffling
+    del shuffle_files, seed
+    print("we have reached the end of this func")
+    dataset = datasets.load_dataset(dataset_name, subset_name)
+    dataset = dataset[split_mapping[split]]
+    print("we are now moving to the end of this func")
+    # dataset = utils.apply_template(dataset, template)
+    return hf_dataset_to_tf_dataset(dataset)
+
+def task_clean(text):
+    # Clean the text according to allowed characters for a task name
+    return re.sub(r"[^\w\d\._]+", "_", text)
+
+def get_task_name(dataset_name, subset_name):
+    return task_clean(dataset_name + (f"_{subset_name}_" if subset_name is not None else "_"))
+
+# ========================= XP3 Training Sets ===========================
+for task_name in constants_xp3.XP3_TRAIN_TASK_SPLITS:
+  subtask_id = task_name.split(":")[-1]
+  ds_name = subtask_id.split("_")[0]
+  subset_name = subtask_id.split("_")[1]
+  if constants_t0.T0_TRAIN_TASK_METADATA[task_name]["in_flan"]:
+    continue
+#   Do not process T0 variants with negative examples.
+#   We still keep the variants of these sets in a different format.
+#   if "_score_eval" in subtask_id:
+#     continue
+#   elif constants_t0.T0_TRAIN_TASK_METADATA[task_name][
+#       "task_type"] == "t0_question_answer":
+#     preprocessors = [functools.partial(prep.t0, multiple_choice=False)]
+#     if constants_t0.T0_TRAIN_TASK_METADATA[task_name]["seq_len"]["max"] == 1:
+#       postprocessors = functools.partial(post.take_first_word)
+#     else:
+#       postprocessors = functools.partial(post.take_first_line)
+#   elif constants_t0.T0_TRAIN_TASK_METADATA[task_name]["task_type"] in [
+#       "t0_multiple_choice", "t0_multiple_choice_separated_options"
+#   ]:
+    preprocessors = [functools.partial(prep.t0, multiple_choice=True)]
+    postprocessors = None
+    # Only include non-deterministic options if they aren't already hard-coded.
+    # if constants_t0.T0_TRAIN_TASK_METADATA[task_name][
+    #     "task_type"] == "t0_multiple_choice_separated_options":
+    #   preprocessors.append(prep.format_options)
+
+  t0_metadata_prep = functools.partial(prep.add_source_info,
+    task_name=subtask_id, task_source="P3")
+  XP3_TASK_CONFIGS[task_name] = TaskConfig(
+      source=seqio.TfdsDataSource(
+          tfds_name=f"huggingface:{ds_name}/{subset_name}",
```

Comment:
From an eye test I think this works: https://www.tensorflow.org/datasets/community_catalog/huggingface seems to have the datasets I checked randomly. 

There's some custom [version picking in xP3](https://github.com/bigscience-workshop/bigscience/blob/aa872e754106f6678e8a9dac8c6962404ba39a6d/data/xp3/prepare_xp3_train.py#L1086) that will need to be brought over. I plan to put that into the metadata and use it here.